armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -259,6 +265,33 @@ def _isTopDummyBlockPresent(self):
                 runLog.error(msg)
                 raise RuntimeError(msg)
 
+    def _checkForBlocksWithoutSolids(self):
+        """
+        Makes sure that there aren't any blocks (other than the top-most dummy block)
+        that are entirely fluid filled, unless all blocks in the assembly are only
+        fluids. The expansion changer doesn't know what to do with such mixed assemblies.
+        """
+        solidCompsInAssem = [
+            c
+            for c in self.linked.a.iterComponents()
+            if not isinstance(c.material, material.Fluid)
+        ]
+        if len(solidCompsInAssem) == 0:
+            return
+
+        for b in self.linked.a[:-1]:
+            # the topmost block has already been confirmed as the dummy block
+            solidCompsInBlock = [
+                c
+                for c in b.iterComponents()
+                if not isinstance(c.material, material.Fluid)
+            ]
+            if len(solidCompsInBlock) == 0:
+                raise InputError(
+                    f"Assembly {self.linked.a} is constructed improperly for use with the axial expansion changer.\n"
+                    "Consider using the assemFlagsToSkipAxialExpansion case setting."
```

Comment:
Let's give a little more detail so users don't have to dig into the source code. E.g. "...is constructed improperly for use with the axial expansion changer; i.e., assemblies cannot have intermediate blocks void of a solid component. Consider using....

Reply:
Well ideally there would be some documentation that outlines how to use the axial expansion capability ;-)

Reply:
Hah! Yeah yeah.... very true....